src/core/border_router/routing_manager.hpp

```diff
@@ -0,0 +1,285 @@
+     * @see PublishOmrPrefix
+     * @see UnpublishOmrPrefix
+     *
+     */
+    void EvaluateRoutingPolicy(void);
+
+    /**
+     * This method evaluates the OMR prefix for the Thread Network.
+     *
+     * @returns  The new OMR prefix should be advertised. An invalid OMR prefix
+     *           means we should no longer publish an OMR prefix in the Thread network.
+     *
+     */
+    Ip6::Prefix EvaluateOmrPrefix(void);
+
+    /**
+     * This method evaluates the on-link prefix for the infra link.
+     *
+     * @returns  The new on-link prefix should be advertised. An invalid on-link prefix
+     *           means we should no longer advertise an on-link prefix on the infra link.
+     *
+     */
+    Ip6::Prefix EvaluateOnLinkPrefix(void);
+
+    /**
+     * This method publishes an OMR prefix in Thread network.
+     *
+     * @param[in]  aOmrPrefix  An OMR prefix.
+     *
+     */
+    otError PublishOmrPrefix(const Ip6::Prefix &aOmrPrefix);
+
+    /**
+     * This method unpublishes an OMR prefix if we have done that.
+     *
+     * @param[in]  aOmrPrefix  An OMR prefix.
+     *
+     */
+    void UnpublishOmrPrefix(const Ip6::Prefix &aOmrPrefix);
+
+    /**
+     * This method starts sending Router Solicitations in random delay
+     * between 0 and kMaxRtrSolicitationDelay.
+     *
+     */
+    void StartRouterSolicitation();
+
+    /**
+     * This method sends Router Solicitation messages to discovery on-link
+     * prefix on infra links.
+     *
+     * @see HandleRouterAdvertisement
+     *
+     */
+    otError SendRouterSolicit(void);
+
+    /**
+     * This method sends Router Advertisement messages to advertise
+     * on-link prefix and route for OMR prefix.
+     *
+     * @param[in]  aNewOmrPrefix  The new OMR prefix to be advertised.
+     *                            An invalid OMR prefix means we should stop advertising OMR prefix.
+     * @param[in]  aOnLinkPrefix  The new on-link prefix to be advertised.
+     *                            An invalid on-link prefix means we should stop advertising on-link prefix.
+     *
+     */
+    void SendRouterAdvertisement(const Ip6::Prefix &aNewOmrPrefix, const Ip6::Prefix &aNewOnLinkPrefix);
+
+    static void HandleRouterAdvertisementTimer(Timer &aTimer);
+    void        HandleRouterAdvertisementTimer(void);
+
+    static void HandleRouterSolicitTimer(Timer &aTimer);
+    void        HandleRouterSolicitTimer(void);
+
+    static void HandleDiscoveredOnLinkPrefixInvalidTimer(Timer &aTimer);
+    void        HandleDiscoveredOnLinkPrefixInvalidTimer(void);
+
+    void HandleRouterSolicit(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
+    void HandleRouterAdvertisement(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
+
+    static TimeMilli GetPrefixExpireTime(uint32_t aValidLifetime);
+
+    const char *GetInfraIfName() const { return otPlatInfraIfGetName(mInfraIfIndex); }
```

Comment:
```suggestion
    const char *GetInfraIfName() const { static const char *name = otPlatInfraIfGetName(mInfraIfIndex); return name; }
```
nit: Reduce the call to `otPlatInfraIfGetName` in case it's costy?

Reply:
removed this API.